cmd/cluster/core/dump.go

```diff
@@ -139,7 +139,21 @@ func NewDumpCommand() *cobra.Command {
 
 func dumpGuestCluster(ctx context.Context, opts *DumpOptions) error {
 	start := time.Now()
-	c, err := util.GetClient()
+	var c client.Client
+	var err error
+
+	if len(opts.ImpersonateAs) > 0 {
+		c, err = util.GetImpersonatedClient(opts.ImpersonateAs)
```

Comment:
Hey @Tafhim, thank you for the PR!

Here, I think, you're assuming that the same user will be set in the Management cluster and the Hosted cluster by default, and I don't think this is the most used case, am I right?.

Maybe another approach could be adding another flag to indicate this new user in the guest cluster? 

WDYT @csrwng @enxebre ?

Reply:
The impersonation is to get the kubeconfig and later to port forward, both are operations done on the management cluster. So it is correct to do the impersonation. We just missed this.